give/src/components/Teaser/Teaser.js

```diff
@@ -0,0 +1,114 @@
+// Based on v2
+// https://www.aemcomponents.dev/content/core-components-examples/library/core-content/teaser.html
+// https://github.com/adobe/aem-core-wcm-components/blob/main/content/src/content/jcr_root/apps/core/wcm/components/teaser/v2/teaser/teaser.html
+
+import { ComponentWrapper, LinkWrapper } from "../shared";
+import { Image } from "../Image/Image";
+import "./Teaser.css";
+import lava1 from "../../assets/lava1.jpeg";
+
+export const Teaser = ({
+  className = "", // string
+  type = "", // string - "featured" | "layered"
+  title = "", // string
+  img = { src: "", alt: "" }, // object with `src` and `alt` keys, each key containing a string
+  pre = "", // string
+  desc = "", // string
+  btns = [], // array of objects with `href` and `text` keys, each key containing a string
+  link = "", // string - URL
+}) => {
+  if (!title) return null;
+
+  const types = ["featured", "layered"];
+  const typeClass = types.includes(type) ? "cru-teaser-" + type : "";
+
+  return (
+    <ComponentWrapper type="teaser" className={`${typeClass} ${className}`}>
+      <LinkWrapper href={link} className="cmp-teaser__link">
+        {/* Content */}
+        <div className="cmp-teaser__content">
+          {/* Pre-title */}
+          {pre && <p className="cmp-teaser__pretitle">{pre}</p>}
+
+          {/* Title */}
+          <h2 className="cmp-teaser__title">
+            <LinkWrapper href={link} className="cmp-teaser__title-link">
+              {title}
+            </LinkWrapper>
+          </h2>
+
+          {/* Description */}
+          {desc && (
+            <div className="cmp-teaser__description">
+              <p>{desc}</p>
+            </div>
+          )}
+
+          {/* Buttons */}
+          {btns.length > 0 && (
+            <div className="cmp-teaser__action-container">
+              {btns.map((btn) => (
```

Comment:
The type would be nice in this map too so someone could reference easier to know what is needed for a button